src/proc/pointcloud.cpp

```diff
@@ -141,7 +141,8 @@ namespace librealsense
                                         sensor_interface & s = dev->get_sensor( x );
                                         for( auto const & sp : s.get_active_streams() )
                                         {
-                                            if( sp->get_stream_type() == RS2_STREAM_COLOR )
+                                            if(( sp->get_stream_type() == RS2_STREAM_COLOR ) &&
+                                               ( _prev_stream_filter.stream == RS2_STREAM_COLOR ))
```

Comment:
Why do we want the prev filter to be color?

Reply:
The way I read it:
* Whenever Color stream is enabled, Color sensor's texel intrinsic and extrinsic is getting used instead of Depth sensor's.
That means that we want to take the color intrinsics only when we have scenario 1: we actually want to use Color.

What I don't quite understand is why that's reflected in `_prev_stream_filter`, but I guess we just have to stare at the code some more :)

Reply:
> The way I read it:
> 
> * Whenever Color stream is enabled, Color sensor's texel intrinsic and extrinsic is getting used instead of Depth sensor's.
> 
> That means that we want to take the color intrinsics only when we have scenario 1: we actually want to use Color.

Yes, @maloel. That's right.

> What I don't quite understand is why that's reflected in _prev_stream_filter, but I guess we just have to stare at the code some more :)

If we look at this function `inspect_other_frame(...)` , user will pass the required frame:
- `Color` frame for scenario 1
- `Colorized depth` frame for scenario 2 
- or even `Infrared` frame also possible I think

Some default intrinsic and extrinsic values are set for the first time at line 191 and 192 in this same function.
Then, later in this callback, if the Color stream is enabled, they are overwritten by Color sensor's intrinsic and extrinsic. 
So, to prevent this overwrite, added a condition to check whether the user want's `Color` stream or not.

Reply:
If we see the flow of the API calls:

Top level API is `map_to(frame mapped)`. refer [rs_processing.h](https://github.com/IntelRealSense/librealsense/blob/development/include/librealsense2/hpp/rs_processing.hpp#L472)
- Here `_stream_filter` will be set as per user's need
  - Color for scenario 1
  - Depth for scenario 2
- Then, `process(...)` function is called
  - which in turn calls `inspect_other_frame(...)` and it sets `_prev_stream_filter`

Reply:
Right, but nothing initialized `_prev_stream_filter`...

Reply:
at line 115 in this same function, it is initialized.

Reply:
But it's used in line 113 😄

Reply:
Looks like, it is getting initialized with default values (`RS2_STREAM_ANY`, `RS2_FORMAT_ANY`, ...) in the constructor `stream_filter` class in [synthetic-stream.h](https://github.com/IntelRealSense/librealsense/blob/development/src/proc/synthetic-stream.h#L90)

Reply:
Hmm right you are.
Then why compare the **previous** stream filter and not simply `_stream_filter`? I think that would be preferable?